pkg/plugin/registry.go

```diff
@@ -0,0 +1,52 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"strings"
+
+	"github.com/conduitio/conduit/pkg/foundation/log"
+)
+
+const builtinPluginPrefix = "builtin:"
+
+// registry is an object that can create new plugin dispensers. We need to use
+// an interface to prevent a cyclic dependency between the plugin package and
+// builtin and standalone packages.
```

Comment:
Consider adding to this comment a summary of what we are defining as builtin and standalone packages. As someone coming into the code, it's not immediately apparent what the distinction is.

Reply:
Done.